radio/src/gui/colorlcd/radio_hardware.cpp

```diff
@@ -330,10 +330,17 @@ void RadioHardwarePage::build(FormWindow * window)
 #if defined(HARDWARE_INTERNAL_MODULE)
   new StaticText(window, grid.getLabelSlot(), TR_INTERNAL_MODULE, 0,
                  COLOR_THEME_PRIMARY1);
-  auto internalModule =
-      new Choice(window, grid.getFieldSlot(1, 0), STR_INTERNAL_MODULE_PROTOCOLS,
-                 MODULE_TYPE_NONE, MODULE_TYPE_COUNT - 1,
-                 GET_SET_DEFAULT(g_eeGeneral.internalModule));
+  auto internalModule = new Choice(window, grid.getFieldSlot(1, 0),
+      STR_INTERNAL_MODULE_PROTOCOLS, MODULE_TYPE_NONE, MODULE_TYPE_COUNT - 1,
+      GET_DEFAULT(g_eeGeneral.internalModule),
+      [=](int moduleType) {
+        if (!isInternalModuleAvailable(moduleType)) {
+          memclear(&g_model.moduleData[INTERNAL_MODULE], sizeof(ModuleData));
```

Comment:
Shoudn't be checked if `g_model.moduleData[INTERNAL_MODULE].type != moduleType` before clearing it? I'm not sure the *set* function is not called even the value has not changed.

Reply:
The set function is still called, but given the second test of isInternalModuleAvailable() 
```
if (g_eeGeneral.internalModule != moduleType) 
return false;
```
since the selected moduleType didn't change from what already set, it correctly skipped the clear. Regardless, that is clearer, doesn't have any extra stuff that could come back to bite me, and basically the same as the B&W test so will do that. :)